llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -17611,6 +17628,8 @@ static SDValue performMulCombine(SDNode *N, SelectionDAG &DAG,
     //     => MV = (add (shl x, M), x); (add (shl MV, N), MV)
     // (mul x, (2^M + 1) * 2^N + 1))
     //     =>  MV = add (shl x, M), x); add (shl MV, N), x)
+    // (mul x, 1 - (1 - 2^M) * 2^N))
+    //     =>  MV = sub (x - (shl x, M)); add (x - (shl x, M))
```

Comment:
Formula looks inconsistent with other formulas.

Reply:
Updated, it is `MV = sub (x - (shl x, M)); add (x - (shl MV, N))`, thanks.